components/authentication/auth.js

```diff
@@ -0,0 +1,19 @@
+import jwt from 'jsonwebtoken'
+import User from '../users/user'
+
+const auth = async (req, res, next) => {
+  try {
+    const token = req.header('Authorization').replace('Bearer ', '')
+    const decoded = jwt.decode(token, process.env.SECRET_KEY)
```

Comment:
Remember to add the `secret` to our environment file and use here, instead of access directly `process.env`